apps/state-demo/src/app/examples/demo-basics/1/Readme.md

```diff
@@ -1,62 +1,264 @@
-# Implementing state, selections and UI interaction
+# Setup a reactive state, its selections and the related UI interactions
 
-0. Implement reactive template helpers (push pipe)
+In this section we start with an imperative code base and will refactor
+its state management to a reactive setup.
+This step will include the components input bindings, child component output as well as global state.
 
-```html
-{{storeList$ | async}}}
-```
+We will ensure automated subscription handling, imperative interaction
+over input bindings and a clean separation of responsibilities.
 
-0.1 finetune with strategies
+---
 
-```html
-{{storeList$ | async: 'local'}}}
+1. Implement RxState Service ComponentState by extending the RxState class.
+
+Let's start with introducing a reactive state to our component.
+This can be done over inheritance, we extend form the state service, or composition, we inject the service in the constructor and add the service to the component `providers` section.
+
+In this article we simply extend from the service.
+The benefit here is we can access the services API directly over `this`. e.g. `this.select('prop')`.
+
+To do so, we have to extend our class and use the already existing `ComponentState` interface:
+
+```typescript
+import { RxState } from '@rx-angular/state';
+
+// Displayed shape of the list item (this is a reduced version of the server object)
+export interface DemoBasicsItem {
+  id: string;
+  name: string;
+}
+
+interface ComponentState {
+  refreshInterval: number;
+  list: DemoBasicsItem[];
+  listExpanded: boolean;
+}
+
+export class DemoBasicsComponent2 extends RxState<ComponentState> {
 ```
 
-1. Implement RxState Service ComponentState by extending the RxState class.
+Also, a `super` in the constructor is needed as we extend from another class.
 
 ```typescript
-export class DemoBasicsComponent2 extends RxState<ComponentState>
+constructor(...) {
+  super();
+}
 ```
 
-1.1) Select component State
+1.1) Select component State, display and initialize it.
```

Comment:
```suggestion
## Select and display state
```

Reply:
this should be swapped with the next step, since you initialize state in the next step. Initializing comes before selecting and displaying, doesn't it?